client/extensions/wp-job-manager/components/settings/pages/index.jsx

```diff
@@ -1,5 +1,128 @@
+import FormSettingExplanation from 'components/forms/form-setting-explanation';
+import PageDropdown from './page-dropdown';
+import QueryPosts from 'components/data/query-posts';
+import SectionHeader from 'components/section-header';
+import { getSitePosts, isRequestingSitePostsForQuery } from 'state/posts/selectors';
+import { getSelectedSiteId } from 'state/ui/selectors';
+
+class Pages extends Component {
+	static propTypes = {
+		isDisabled: PropTypes.bool,
+		isRequesting: PropTypes.bool,
+		pages: PropTypes.array,
+		siteId: PropTypes.number,
+		translate: PropTypes.func,
+	};
+
+	render() {
+		const {
+			isRequesting,
+			pages,
+			siteId,
+			translate,
+		} = this.props;
+
+		const isDisabled = this.props.isDisabled || isRequesting;
+
+		return (
+			<div>
+				<QueryPosts siteId={ siteId } query={ { type: 'page' } } />
+
+				<form>
+					<FormSection name="pages">
+						<SectionHeader label={ translate( 'Pages' ) }>
+							<FormButton compact
+								disabled={ isDisabled } />
+						</SectionHeader>
+						<Card>
+							<FormFieldset>
+								<FormLabel htmlFor="submitFormPage">
+									{ translate( 'Submit Job Form Page' ) }
+								</FormLabel>
+								<FormSettingExplanation>
+									{ translate( 'Select the page where you\'ve used the [submit_job_form] shortcode. ' +
+										'This lets the plugin know the location of the form.' ) }
+								</FormSettingExplanation>
+								<PageDropdown
+									disabled={ isDisabled }
+									id="submitFormPage"
+									name="submitFormPage"
+									pages={ pages } />
+							</FormFieldset>
+
+							<FormFieldset>
+								<FormLabel htmlFor="dashboardPage">
+									{ translate( 'Job Dashboard Page' ) }
+								</FormLabel>
+								<FormSettingExplanation>
+									{ translate( 'Select the page where you\'ve used the [job_dashboard] shortcode. ' +
+										'This lets the plugin know the location of the dashboard.' ) }
+								</FormSettingExplanation>
+								<PageDropdown
+									disabled={ isDisabled }
+									id="dashboardPage"
+									name="dashboardPage"
+									pages={ pages } />
+							</FormFieldset>
+
+							<FormFieldset>
+								<FormLabel htmlFor="listingsPage">
+									{ translate( 'Job Listings Page' ) }
+								</FormLabel>
+								<FormSettingExplanation>
+									{ translate( 'Select the page where you\'ve used the [jobs] shortcode. ' +
+										'This lets the plugin know the location of the job listings page.' ) }
+								</FormSettingExplanation>
+								<PageDropdown
+									disabled={ isDisabled }
+									id="listingsPage"
+									name="listingsPage"
+									pages={ pages } />
+							</FormFieldset>
+						</Card>
+					</FormSection>
+				</form>
+			</div>
+		);
+	}
+}
+
+const connectComponent = connect(
+	( state ) => {
+		const siteId = getSelectedSiteId( state );
+
+		return {
+			isRequesting: isRequestingSitePostsForQuery( state, siteId, { type: 'page' } ),
```

Comment:
Might be worth defining a module-level `query` const for use here and with `QueryPosts`.